src/tests/string/test_string_functions.f90

```diff
@@ -52,6 +56,182 @@ subroutine test_reverse_string
+        call check(slice(test_string, 7, 7, 3) == "g", &
+                    'function slice failed', warn=.false.)
+        call check(slice(test_char, 7, -10) == "gfedcba", &
+                    'function slice failed', warn=.false.)
+        call check(slice(test_string, 500, 22) == "zyxwv", &
+                    'function slice failed', warn=.false.)
+        call check(slice(test_char, 50, 27) == "", &
+                    'function slice failed', warn=.false.)
+        call check(slice(test_string, -20, -200) == "", &
+                    'function slice failed', warn=.false.)
+        call check(slice(test_char, first=0, stride=-1) == "", &
+                    'function slice failed', warn=.false.)
+        call check(slice(test_string, last=27, stride=-2) == "", &
+                    'function slice failed', warn=.false.)
+        call check(slice(test_char, first=27, stride=2) == "", &
+                    'function slice failed', warn=.false.)
+        call check(slice(test_string, -500, 500) == "abcdefghijklmnopqrstuvwxyz", &
+                    'function slice failed', warn=.false.)
+
+        test_string = ""
+        test_char = ""
+        call check(slice(test_string, 2, 16, 3) == "", &
+                    'function slice failed', warn=.false.)
+        call check(slice(test_char, 2, 16, 3) == "", &
+                    'function slice failed', warn=.false.)
+
+    end subroutine test_slice_string
+
+    subroutine test_slice_gen
+        character(len=*), parameter :: test = &
+            & "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
+        integer :: i, j, k
+        integer, parameter :: offset = 3
+
+        do i = 1 - offset, len(test) + offset
+            call check_slicer(test, first=i)
+        end do
+
+        do i = 1 - offset, len(test) + offset
+            call check_slicer(test, last=i)
+        end do
+
+        do i = -len(test) - offset, len(test) + offset
+            call check_slicer(test, stride=i)
+        end do
+
+        do i = 1 - offset, len(test) + offset
+            do j = 1 - offset, len(test) + offset
+                call check_slicer(test, first=i, last=j)
+            end do
+        end do
+
+        do i = 1 - offset, len(test) + offset
+            do j = -len(test) - offset, len(test) + offset
+                call check_slicer(test, first=i, stride=j)
+            end do
+        end do
+
+        do i = 1 - offset, len(test) + offset
+            do j = -len(test) - offset, len(test) + offset
+                call check_slicer(test, last=i, stride=j)
+            end do
+        end do
+
+        do i = 1 - offset, len(test) + offset
+            do j = 1 - offset, len(test) + offset
+                do k = -len(test) - offset, len(test) + offset
+                    call check_slicer(test, first=i, last=j, stride=k)
+                end do
+            end do
+        end do
+    end subroutine test_slice_gen
```

Comment:
This makes it even harder to see what the expected behavior is supposed to be. The goal of a good test suite is to serve as an example of how to use the code, and a definition of it's expected behavior. You should go back to the specific tests you had and just give them more meaningful descriptions. The message from a failing test should be a hint as to what aspect of the code is not working correctly. Something as vague as "it failed" doesn't accomplish that. Ideally, I should be able to read the test suite alone and understand the expected behavior.